feat(source-google-directory): add `id` and `email` field to `group_members` stream

KyloJorgensen opened the pull request:
# chore(source-google-directory): update version to 0.2.45 and enrich group member data with group context

- Bump version in pyproject.toml and metadata.yaml to 0.2.45
- Enhance GroupMembersAPI to include group_email and group_id in the member data
- Update group_members.json schema to reflect new fields

## What

<!--
* Describe what the change is solving. Link all GitHub issues related to this change.
-->

This PR addresses a limitation in the Google Directory connector where the `group_members` stream was missing crucial group context. Previously, when fetching group members via the Google Directory API, the returned member records did not include information about which group they belonged to, making it impossible for downstream systems to reconstruct group membership relationships.

**Problem:**

- Group members were returned without group identifiers
- Downstream data warehouses and destinations couldn't join member data with group data
- Group membership relationships were lost during data synchronization

## How

<!--
* Describe how code changes achieve the solution.
-->

**Code Changes:**

1. **Enhanced GroupMembersAPI logic** (`source_google_directory/api.py`):

   - Modified `GroupMembersAPI.list()` method to iterate through groups first
   - For each group, fetch its members and enrich each member record with `group_email` and `group_id` fields
   - Preserves all original member fields while adding group context

2. **Updated schema definition** (`source_google_directory/schemas/group_members.json`):

   - Added `group_email` field: "The email address of the group this member belongs to"
   - Added `group_id` field: "The unique identifier of the group this member belongs to"

3. **Comprehensive: Test
   - Tests verify group context is properly added while maintaining backward compatibility

4. **Version and documentation updates**:
   - Bumped version from 0.2.44 to 0.2.45 (patch-level change)
   - Updated changelog in connector documentation

## Review guide

<!--
1. `x.py`
2. `y.py`
-->

1. **`source_google_directory/api.py`** - Core logic changes in `GroupMembersAPI.list()` method
2. **`source_google_directory/schemas/group_members.json`** - Schema additions for new fields
3. **`source-google-directory/acceptance-test-config.yml`** - Comprehensive test suite for the new functionality
4. **`pyproject.toml`** - Version bump to 0.2.45
5. **`metadata.yaml`** - Docker image tag update to 0.2.45
6. **`docs/integrations/sources/google-directory.md`** - Changelog entry for the new version

## User Impact

<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them.
-->

**Positive Impact:**

- ✅ **Enhanced Data Relationships**: Users can now properly join group member data with group information in their data warehouse
- ✅ **Better Analytics**: Group membership analytics become possible with full context preservation
- ✅ **Downstream Compatibility**: Destinations like Convex, BigQuery, etc. can now properly reconstruct group hierarchies
- ✅ **Backward Compatible**: All existing member fields are preserved, no breaking changes

**Before:**

```json
{
  "id": "member@example.com",
  "email": "member@example.com",
  "role": "MEMBER",
  "type": "USER"
}
```

**After:**

```json
{
  "id": "member@example.com",
  "email": "member@example.com",
  "role": "MEMBER",
  "type": "USER",
  "group_email": "team@example.com",
  "group_id": "team@example.com"
}
```

**Potential Side Effects:**

- Destinations may need to update their schema to capture the new `group_email` and `group_id` fields
- No negative functional impact - purely additive enhancement

## Can this PR be safely reverted and rolled back?

<!--
* If unsure, leave it blank.
-->

- [x] YES 💚
- [ ] NO ❌

**Why it's safe to revert:**

- All changes are additive (no existing functionality removed)
- Schema changes are backward compatible
- Version bump is incremental (patch-level)
- No breaking API changes
- Rollback would simply remove the new fields, returning to previous behavior

---

## Technical Implementation Details

### Code Changes Summary

**File: `airbyte-integrations/connectors/source-google-directory/source_google_directory/api.py`**

```python
# Before
def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
    groups = GroupsAPI(self._api)
    for group in groups.list():
        params = {"groupKey": group["id"]}
        yield from self.read(partial(self._api_get, resource="members"), params=params)

# After
def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
    groups = GroupsAPI(self._api)
    for group in groups.list():
        group_email = group.get("email")
        group_id = group.get("id")
        params = {"groupKey": group_id}

        # Get members for this group and enrich with group context
        for member in self.read(partial(self._api_get, resource="members"), params=params):
            # Enrich member with group context
            member["group_email"] = group_email
            member["group_id"] = group_id
            yield member
```

**File: `airbyte-integrations/connectors/source-google-directory/source_google_directory/schemas/group_members.json`**

```json
{
  "properties": {
    // ... existing fields ...
    "group_email": {
      "type": ["null", "string"],
      "description": "The email address of the group this member belongs to"
    },
    "group_id": {
      "type": ["null", "string"],
      "description": "The unique identifier of the group this member belongs to"
    }
  }
}
```

### Testing Coverage

- ✅ **Group context enrichment functionality**
- ✅ **Original field preservation**
- ✅ **Edge cases (empty groups, no members, missing API keys)**
- ✅ **Schema validation**
- ✅ **Backward compatibility**

KyloJorgensen committed:
kylojorgensen/group-member-group-id-and-group-email

chore(source-google-directory): update version to 0.2.45 and enrich group member data with group context

- Bump version in pyproject.toml and metadata.yaml to 0.2.45
- Enhance GroupMembersAPI to include group_email and group_id in the member data
- Update group_members.json schema to reflect new fields

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=62526) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=62526) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=62526) it.</sub>

marcosmarxm committed:
update doc

marcosmarxm reviewed (dismissed):
Seems correct @KyloJorgensen need you to sign the CLA to continue the review.

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

KyloJorgensen committed:
remove unnecessary white space in GroupMembersAPI class

KyloJorgensen committed:
feat(source-google-directory): add group_id and group_email fields to configured_catalog.json

KyloJorgensen committed:
refactor(source-google-directory): update acceptance test configuration to use 'acceptance_tests' key

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

cgardens commented:
@KyloJorgensen I should be able to merge this once this ci error is addressed

```
source-google-directory - ❌ Failed - Connector license in metadata.yaml and pyproject.toml file must match: Connector is licensed under MIT in pyproject.toml, but licensed under ELv2 in metadata.yaml. These two files have to be consistent.
Error: 1 checks failed
```

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

KyloJorgensen committed:
fix(source-google-directory): align license in pyproject.toml with metadata.yaml

Update license from MIT to ELv2 in pyproject.toml to match the license
specified in metadata.yaml, resolving connector build validation error.

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

KyloJorgensen commented:
> @KyloJorgensen I should be able to merge this once this ci error is addressed
> 
> ```
> source-google-directory - ❌ Failed - Connector license in metadata.yaml and pyproject.toml file must match: Connector is licensed under MIT in pyproject.toml, but licensed under ELv2 in metadata.yaml. These two files have to be consistent.
> Error: 1 checks failed
> ```

@cgardens Done

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

cgardens commented:
note to self: referred to internal team -- extensibility

KyloJorgensen committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

DanyloGL committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email

DanyloGL committed:
Merge branch 'master' into kylojorgensen/group-member-group-id-and-group-email
